lib/Conversion/StructuredToMemref/StructuredToMemref.cpp

```diff
@@ -762,19 +761,43 @@ struct ScalarAddptrConverter : public OpConversionPattern<triton::AddPtrOp> {
         cast<triton::PointerType>(op.getPtr().getType()).getPointeeType();
     auto memrefType = MemRefType::get({1}, elemType, layout);
 
-    auto extractMetadataOp =
-        rewriter.create<memref::ExtractStridedMetadataOp>(loc, ptr);
-    auto base = extractMetadataOp.getBaseBuffer();
-    auto offset = extractMetadataOp.getOffset();
+    auto ptr = adaptor.getPtr();
+    auto definingOp = ptr.getDefiningOp();
+
+    if (definingOp) {
+      auto reinterpretCast = cast<memref::ReinterpretCastOp>(definingOp);
+      auto base = reinterpretCast.getSource();
+      auto offsets = reinterpretCast.getMixedOffsets();
+      assert(offsets.size() == 1);
```

Comment:
is this an invariant or should you return failure()  here if offsets != 1

Reply:
It is an invariant because the reinterpret_casts are always created by us. And since we're working with scalars only, the offset should always have size == 1